src/service/url-expander/parser.js

```diff
@@ -0,0 +1,227 @@
+import {rethrowAsync} from '../../log';
+import {encodeValue} from '../url-replacements-impl';
+
+/** Rudamentary parser to handle nested Url replacement. */
+export class Parser {
+
+  /**
+   * Link this instance of parser to the calling UrlReplacment
+   * @param {VariableSource} variableSource the keywords to replace
+   */
+  constructor(variableSource) {
+    this.variableSource_ = variableSource;
+  }
+
+  /**
+   * Takes array of all matching variable keyword locations and returns only those that we
+   * want to evaluate. In the case of overlapping keywords it choose the longer. It also
+   * excludes protected keywords.
+   * @param {!Array<Object<string, string|number>>} matches array of objects representing
+   *  matching keywords
+   * @param {string} url The url to be expanded.
+   * @return {!Object<string, string|number>}
+   */
+  eliminateOverlaps_(matches, url, opt_whiteList) {
+    if (!matches.length) { return null; }
+    // longest keywords have priority over shorter
+    // how should we prioritize if same length??
```

Comment:
Right now this is kind of arbitrary, should we have a way to decide if keywords are same length?